src/arfsdao.ts

```diff
@@ -196,6 +199,245 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
 
 		return driveBuilder.build();
 	}
+
+	async getPublicFolder(folderId: string): Promise<ArFSPublicFolder> {
+		const gqlQuery = buildQuery([{ name: 'Folder-Id', value: folderId }]);
+
+		const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+
+		const { data } = response.data;
+		const { transactions } = data;
+		const { edges } = transactions;
+
+		if (!edges.length) {
+			throw new Error(`Public folder with Folder ID ${folderId} not found!`);
+		}
+
+		const folderBuilder = new ArFSPublicFolderBuilder();
+
+		const { node } = edges[0];
+		const { tags } = node;
+		tags.forEach((tag: GQLTagInterface) => {
+			const key = tag.name;
+			const { value } = tag;
+			switch (key) {
+				case 'App-Name':
+					folderBuilder.appName = value;
+					break;
+				case 'App-Version':
+					folderBuilder.appVersion = value;
+					break;
+				case 'ArFS':
+					folderBuilder.arFS = value;
+					break;
+				case 'Content-Type':
+					folderBuilder.contentType = value as ContentType;
+					break;
+				case 'Drive-Id':
+					folderBuilder.driveId = value;
+					break;
+				case 'Entity-Type':
+					folderBuilder.entityType = value as EntityType;
+					break;
+				case 'Unix-Time':
+					folderBuilder.unixTime = +value;
+					break;
+				case 'Parent-Folder-Id':
+					folderBuilder.parentFolderId = value;
+					break;
+				case 'Folder-Id':
+					folderBuilder.entityId = value;
+					break;
+				default:
+					break;
+			}
+		});
+
+		// Get the drives transaction ID
+		folderBuilder.txId = node.id;
+
+		if (folderBuilder.txId) {
+			const txData = await this.arweave.transactions.getData(folderBuilder.txId, { decode: true });
+			const dataString = await Utf8ArrayToStr(txData);
+			const dataJSON = await JSON.parse(dataString);
+
+			// Get the drive name and root folder id
+			folderBuilder.name = dataJSON.name;
+		}
+
+		return folderBuilder.build();
+	}
+
+	async getChildrenOfFolderTxIds(folderId: FolderID): Promise<string[]> {
```

Comment:
Make sure this function is actually used. Remove otherwise.

Reply:
Removed a909772f4b8a994a7297a68f9abfff6aa69000c4

Reply:
> Removed a909772

Appears to still be in the diff

Reply:
That's true. Here it should be removed: 51ed344685124185f76541a0919b68161b085739